pipeline/extract.py

```diff
@@ -0,0 +1,126 @@
+"""Extracts relevant data from S3 Bucket"""
+
+import os
+import logging
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from boto3 import client
+from botocore.config import Config
+import pandas as pd
+from dotenv import load_dotenv
+from pytrends.request import TrendReq
+
+load_dotenv(".env")
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s',
+    handlers=[
+        logging.StreamHandler(),
+    ]
+)
+
+def s3_connection() -> client:
+    """Connects to an S3"""
+    config = Config(
+        connect_timeout=5,
+        read_timeout=10,
+        retries={
+            'max_attempts': 3,
+            'mode': 'standard'
+        },
+        max_pool_connections=50
+    )
+    s3 = client(
+        "s3",
+        aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+        aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+        config=config
+    )
+    return s3
+
+def initialize_trend_request():
+    """Initialize and return a TrendReq object."""
+    return TrendReq()
+
+def fetch_file_content(s3: client, file_name: str, topic: list[str]) -> dict:
+    """Checks files from S3 fir keywords and returns relevant data if keyword is found"""
+    try:
+        file_obj = s3.get_object(Bucket=os.environ.get("S3_BUCKET_NAME"), Key=file_name)
+        file_content = file_obj['Body'].read().decode('utf-8')
+        for keyword in topic:
+            if keyword in file_content:
+                logging.info("Keyword '%s' found in %s", keyword, file_name)
+                return {
+                    "Text": file_content,
+                    "Keyword": keyword,
+                }
+    except FileNotFoundError as e:
+        logging.error("No files found in S3: %s", e)
+    return None
+
+def access_bluesky_files(s3: client, topic: list[str]) -> list[str]:
```

Comment:
could we separate this into 2 functions. One for listing the s3 objects and one for accessing bluesky files.

Reply:
That we can